sycl/doc/SYCLPluginInterface.md

```diff
@@ -0,0 +1,178 @@
+# The SYCL Runtime Plugin Interface.
+
+
+## Overview
+The SYCL Runtime Plugin Interface (PI) is the interface layer between
+device-agnostic part of the SYCL runtime and the device-specific runtime layers
+which control execution on the device. It employs the “plugin” mechanism to bind
+to the device specific runtime layers similarly to what is used by libomptarget
+or OpenCL.
+
+![PI in SYCL runtime architecture](images/SYCL_RT_arch.svg)
+
+The plugin interface and the discovery process behind it allows to dynamically
+plug in implementations based on OpenCL and “native” runtime for a particular
+device – such as OpenCL for
+FPGA devices or native runtimes for GPUs. Implementations of the PI are
+“plugins” - dynamic libraries or shared objects which expose a number of entry
+points implementing the PI interface. The SYCL runtime collects those function
+pointers into a PI interface dispatch table - one per plugin - and uses this
+table to dispatch to the device(s) covered by the corresponding plugin. Further
+sections give complete descriptions of all entry points as well as the dispatch
+table.
+
+PI is based on a subset of OpenCL 1.2 runtime specification, it follows its
+platform, execution and memory models in all aspects except those explicitly
+mentioned in this document. A part of PI API types and functions have exact
+matches in OpenCL. Whenever there is such a match, the semantics also fully
+matches unless the differences are explicitly specified in this document. While
+PI has roots in OpenCL, it does have many differences, and the gap is likely
+to grow, for example in the areas of memory model and management, program
+management.
```

Comment:
After the (welcomed) latest changes to the naming scheme, is not so clear anymore which functions map 1:1 to OpenCL. 
Maybe the document could contain the mapping down below?

Reply:
I'd expect (at least part of) PI<->OpenCL mapping is to evolve, so I suggest that instead every function in the pi.h is commented whether it has direct OpenCL couterpart or not.